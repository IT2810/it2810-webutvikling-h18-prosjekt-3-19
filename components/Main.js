import React, { Component } from 'react';
import {
    View,
    Text,
    StyleSheet,
    TextInput,
    ScrollView,
    TouchableOpacity,
    Alert,
    AsyncStorage,
} from 'react-native';
import Todo from './Todo';

export default class Completed extends Component {
    constructor(props){
        super(props);
        this.state = {
            todoArray: [],
            todoText: '',

            JsonDB: {
              'todos':[],
              'totalcount': 0,
              'todaycount': 0,
              'today': '2000/01/01',
            },
        };
    }

    render() {
        let todos = this.state.todoArray.map((val, key)=>{
            return <Todo key={key} keyval={key} val={val}
                    deleteMethod={()=>this.deleteTodo(key)}/>
        });
        return (
            <View style={styles.todo}>
                <View>

                        <TextInput 
                            style={styles.textInput}
                            placeholder='Write Todo Here!'
                            onChangeText={(todoText)=> this.setState({todoText})}
                            value={this.state.todoText}
                            placeholderTextColor='black'
                            underlineColorAndroid='transparent'>
                        </TextInput>

                        {/* DEBUG */}
                        <Text>{ this.state.debug }</Text>

                        <TouchableOpacity onPress={ this.addTodo.bind(this) } style={styles.addButton}>
                            <Text style={styles.addButtonText}>+</Text>
                        </TouchableOpacity>
                </View>
                <ScrollView style={styles.todoContainer}>
                    {todos}
                </ScrollView>
            </View>
        );
    }

    // When app starts get JSON from storage
    componentDidMount() {
      this.getJSON();
    }

    getJSON = async () => {
      try {
        const value = await AsyncStorage.getItem('@ToDoStore:JSON');
        if (value !== null) {
          // We have data!!
          this.setState({
            JsonDB: JSON.parse(value),
            debug: value,
          })
          this.parseDB(JSON.parse(value));
        }
      } catch (error) {
        Alert.alert('Error retrieving data');
      }
    }

    parseDB(data) {
      let todos = data.todos

       this.setState({
          debug: JSON.stringify(todos[1]),
        })

      // for (let i = 0; i < todos.length; i++) { 
      //   this.state.todoArray.push({
      //     'date': date_,
      //     'todo': todo_,
      //   });
      // }
    }

    // When a new todo is added
    addTodo(){
        if(this.state.todoText){
            var d = new Date();

            let date_ = d.getFullYear() + "/"+(d.getMonth()+1) + "/"+ d.getDate();
            let todo_ = this.state.todoText;

            this.state.todoArray.push({
                'date': date_,
                'todo': todo_,
            });

            this.setState({ todoArray: this.state.todoArray });
            this.setState({ todoText:'' });
            
            this.saveTodo(todo_, date_);
        }
    }

    saveTodo(content, today) {

      let newTodo = {
        'text': content,
        'date': today,
      }

      let JSONdata = this.state.JsonDB;

      JSONdata.todos.push(newTodo);

      this.setState({
        JsonDB: JSONdata,
      });

      this._storeEntry('@ToDoStore:JSON', JSONdata);

    }

    deleteTodo(key){
        this.state.todoArray.splice(key, 1);
        this.setState({todoArray: this.state.todoArray});
    }

    _storeEntry = async (key, data) => {
      try {
        await AsyncStorage.setItem('@ToDoStore:JSON', JSON.stringify(data));
      } catch (error) {
        Alert.alert('Error saving data');
      }
    }


}
const styles = StyleSheet.create({
    todo: {
        flex: 1,
    },
    todoContainer: {
        position: 'relative',
        flex: 2,
    },
    textInput: {
        alignSelf: 'stretch',
        color: '#000',
        padding: 20,
        backgroundColor: 'white',
        borderTopWidth:2,
        borderTopColor: '#ededed'     
    },
    addButton: {
        position: 'absolute',
        zIndex: 11,
        right: 10,
        top: 10,
        backgroundColor: '#E91E63',
        width: 40,
        height: 40,
        borderRadius: 5,
        alignItems: 'center',
        justifyContent: 'center',
    },
    addButtonText: {
        color: '#fff',
        fontSize: 24,
    }
});
