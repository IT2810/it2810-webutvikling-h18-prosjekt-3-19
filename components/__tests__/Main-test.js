import React from "react";
import Main from "../Main";
import renderer from "react-test-renderer";
/*
describe("Testing Main", () => {
  const tree = renderer.create(<Main />).getInstance();

  const initialState = {
    noteArray: [],
    noteText: "initialText"
  };
*/
test("renders correctly", () => {
  const tree = renderer.create(<Main />);
  expect(tree.toJSON()).toMatchSnapshot();
}); /*

  describe("Checking initial State", () => {
    test("add new Note is closed", () => {
      expect(tree.toTree().rendered.state.addNote).toBe(false);
    });

    test("input Text is empty", () => {
      expect(tree.toTree().rendered.instance.state.noteText).toBe("");
    });
  });

  describe("Note functions", () => {
    test("add new note successfully", () => {
      tree.toTree().rendered.instance.addNote();
      expect(tree.toTree().rendered.instance.state.todoList).toHaveLength(1);
    });

    test("delete note successfully", () => {
      tree.toTree().rendered.instance.deleteNote(0);
      expect(tree.toTree().rendered.instance.state.todoList).toHaveLength(0);
    });
  });
});*/
