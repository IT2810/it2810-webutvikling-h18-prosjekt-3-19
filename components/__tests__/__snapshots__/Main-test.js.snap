// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#E91E63",
        "borderBottomColor": "#ddd",
        "borderBottomWidth": 10,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 18,
          "padding": 26,
        }
      }
    >
      - Todo  -
    </Text>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "zIndex": 10,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Add Todo!"
      placeholderTextColor="white"
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#252525",
          "borderTopColor": "#ededed",
          "borderTopWidth": 2,
          "color": "#fff",
          "padding": 20,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#E91E63",
        "borderRadius": 35,
        "bottom": 90,
        "elevation": 8,
        "height": 70,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 20,
        "width": 70,
        "zIndex": 11,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 24,
        }
      }
    >
      +
    </Text>
  </View>
</View>
`;
